python/nx-cugraph/nx_cugraph/algorithms/link_analysis/hits_alg.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import networkx as nx
+import numpy as np
+import pylibcugraph as plc
+
+from nx_cugraph.convert import _to_graph
+from nx_cugraph.utils import _dtype_param, _get_float_dtype, networkx_algorithm
+
+__all__ = ["hits"]
+
+
+@networkx_algorithm(
+    extra_params={
+        'weight : string or None, optional (default="weight")': (
+            "The edge attribute to use as the edge weight."
+        ),
+        **_dtype_param,
+    }
+)
+def hits(
+    G,
+    max_iter=100,
+    tol=1.0e-8,
+    nstart=None,
+    normalized=True,
+    *,
+    weight="weight",
+    dtype=None,
+):
+    G = _to_graph(G, weight, np.float32)
+    if len(G) == 0:
+        return {}, {}
+    if dtype is not None:
+        dtype = _get_float_dtype(dtype)
+    elif weight in G.edge_values:
+        dtype = _get_float_dtype(G.edge_values[weight].dtype)
+    else:
+        dtype = np.float32
+    if nstart is not None:
+        nstart = G._dict_to_nodearray(nstart, 0, dtype)
+    if max_iter <= 0:
+        if nx.__version__[:3] <= "3.2":
+            raise ValueError("`maxiter` must be a positive integer.")
+        raise nx.PowerIterationFailedConvergence(max_iter)
+    try:
+        node_ids, hubs, authorities = plc.hits(
+            resource_handle=plc.ResourceHandle(),
+            graph=G._get_plc_graph(weight, 1, dtype, store_transposed=True),
+            tol=tol,
+            initial_hubs_guess_vertices=None,
+            initial_hubs_guess_values=nstart,
```

Comment:
Have you tested this with `nstart` passed in?  I'm thinking you need to also pass a vertices array (as you mentioned in pagerank below) based on [this](https://github.com/rapidsai/cugraph/blob/f4bcdc2667a15e2e5031987550322a9d29d8f713/python/pylibcugraph/pylibcugraph/hits.pyx#L121-L123).

Reply:
Yes, this functions appears to work fine with only `initial_hubs_guess_values` provided.

Reply:
So I just took a look at the HITS CAPI and the reason it is not failing is simply because both the initial_hubs_guess vertices and values are ignored when the former is [None](https://github.com/rapidsai/cugraph/blob/branch-23.12/cpp/src/c_api/hits.cpp#L116)